vkcs/publicdns_record.go

```diff
@@ -0,0 +1,56 @@
+package vkcs
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+const (
+	recordTypeA     = "A"
+	recordTypeAAAA  = "AAAA"
+	recordTypeCNAME = "CNAME"
+	recordTypeMX    = "MX"
+	recordTypeNS    = "NS"
+	recordTypeSRV   = "SRV"
+	recordTypeTXT   = "TXT"
+)
+
+func publicDNSRecordStateRefreshFunc(client publicDNSClient, zoneID string, id string, recordType string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		record, err := recordGet(client, zoneID, id, recordType).Extract()
+
+		if err != nil {
+			if _, ok := err.(gophercloud.ErrDefault404); ok {
+				return record, recordStatusDeleted, nil
+			}
+			return nil, "", err
+		}
+
+		return record, recordStatusActive, nil
+	}
+}
+
+func publicDNSRecordParseZoneID(dns string) (string, error) {
+	dnsParts := strings.Split(dns, "/") // /v2/dns/<zone-uuid>
+	if len(dnsParts) != 4 {
+		return "", fmt.Errorf("unable to determine vkcs_publicdns_record zone ID from raw DNS: %s", dns)
+	}
+	zoneID := dnsParts[3]
+	return zoneID, nil
+}
+
+func publicDNSRecordParseID(id string) (string, string, string, error) {
+	parts := strings.Split(id, "/")
```

Comment:
so we need similar comment here with id format

Reply:
Added